File: Minimum_Absolute_Difference_in_BST/getMinimumDifference.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int getMinimumDifference(TreeNode root) {
        prev = null;
        min_diff = Integer.MAX_VALUE;
        inorder(root);
        return min_diff;
    }
    
    private void inorder(TreeNode root) {
        if (root == null) return;
        inorder(root.left);
        if (prev != null) min_diff = Math.min(min_diff, root.val - prev);
        prev = root.val;
        inorder(root.right);
    }
    
    private Integer prev;
    private int min_diff;
}

File: Minimum_Absolute_Difference_in_BST/getMinimumDifference.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def getMinimumDifference(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        def inorder(root):
            if not root: return
            inorder(root.left)
            if self.prev is not None: self.min_diff = min(self.min_diff, root.val - self.prev)
            self.prev = root.val
            inorder(root.right)
                
        self.prev = None
        self.min_diff = float("inf")
        inorder(root)
        return self.min_diff
